src/engine/controls/cuecontrol.cpp

```diff
@@ -1647,8 +1647,9 @@ double CueControl::quantizeCuePoint(double position, QuantizeMode mode) {
         // Quantize to previous beat, fall back to next beat.
         return prevBeat != -1.0 ? prevBeat : (nextBeat != -1.0 ? nextBeat : position);
     } else if (mode == QuantizeMode::NextBeat) {
-        // Quantize to next beat, fall back to previous beat.
-        return nextBeat != -1.0 ? nextBeat : (prevBeat != -1.0 ? prevBeat : position);
+        // use current position if we are already exactly on the grid,
+        // otherwise quantize to next beat, fall back to previous beat.
+        return prevBeat == position ? position : (nextBeat != -1.0 ? nextBeat : (prevBeat != -1.0 ? prevBeat : position));
```

Comment:
These nested ternary operators are hard to read. Please use `if` blocks instead.

Reply:
I agree. I just thought they were nested ternary operators for a reason so I kept them. I'll rewrite both cases with if statements.